src/main/java/com/google/sps/data/SingleplayerProgress.java

```diff
@@ -17,55 +17,56 @@
 import java.util.ArrayList;
 
 /**
-* Represents the static data containing progress a user has made in a game.
+* Represents the static data containing progress a single user has made in a game.
 */
-public class Progress {
-    private final String progressID;
+public class SingleplayerProgress {
+    private final String userID;
+    private final String gameID;
+    private final Coordinates location;
     private final ArrayList<String> hintsFound;
-    private final ArrayList<String> players;
     private final String stageID;
-    private final String gameID;
-    private final boolean gameOver;
 
     /**
-    * Builder class used to create Progress objects.
+    * Builder class used to create SingleplayerProgress objects.
     */
     public static class Builder {
         // Required parameters
-        private final String progressID;
+        private final String userID;
+        private final String gameID;
 
         // Optional parameters - initialized to default values
+        private Coordinates location;
         private ArrayList<String> hintsFound = new ArrayList<>();
         private ArrayList<String> players = new ArrayList<>();
         private String stageID = "N/A";
```

Comment:
Shouldn't we initialize it to something random? Having "N/A" may be a problem if there are multiple stages with "N/A" and then someone tries to retrieve the stage but there are multiple stages with that ID

Reply:
A random ID would have similar issues because a stage with that ID wouldn't even exist. A bug where you're trying to access an id that is "N/A" would be easier to track down.

Reply:
I see, that makes sense. Maybe is there a way to distinguage like N/A + gameID?

Reply:
I don't see why you would need to distinguish between two IDs that are both invalid?

Reply:
Well say u are debugging it and u have N/a but many stages have N/A how would u specify which one it is?

Reply:
I think in the end it is just based off implementation and this would be fine in the end.

Reply:
Each progress object only contains a single stageID. Also putting a random ID at the end of the "N/A" wouldn't tell us any new information. Currently, we could just imagine that it's there to get the same effect.

Reply:
I didn't notice each progress only contains one stageID, in this case I think you're right and it wouldn't matter.